pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1760,7 +1760,9 @@ private void checkConsumedLedgers() {
             if (t instanceof PersistentTopic) {
                 Optional.ofNullable(((PersistentTopic) t).getManagedLedger()).ifPresent(
                         managedLedger -> {
-                            managedLedger.trimConsumedLedgersInBackground(Futures.NULL_PROMISE);
+                            if (!managedLedger.maybeUpdateCursorBeforeTrimmingConsumedLedger()){
```

Comment:
can we add a comment here please, to explain why we are skipping the execution ?

Reply:
add comment to explain the reason, Please take again look, Thx.